### PR TITLE
Implement InsnDetail (+ associated types) for sysz

### PR DESCRIPTION
**Pull request by GarntS**
This PR implements `InsnDetail`, `SysZOperand`, etc.  to enable instruction details for sysz. I used the RISC-V implementation as a template, so the code style should follow the rest of the library's conventions.

### CODE REVIEW

File: capstone-rs/src/instruction.rs [hunk 1]
```diff
@@ -413,6 +413,7 @@ impl<'a> InsnDetail<'a> {
                             $detail($insn_detail(unsafe { &self.0.__bindgen_anon_1.$arch }))
                         }
                     )*
+                    #[allow(unreachable_patterns)]
```
Comment (tmfink) [inline capstone-rs/src/instruction.rs:416]: you can just delete this catch-all pattern arm

File: capstone-rs/src/arch/sysz.rs [hunk 2]
```diff
@@ -1,3 +1,106 @@
 //! Contains sysz-specific types
```
Comment (tmfink) [inline capstone-rs/src/arch/sysz.rs:1]: Please add a `test_arch_sysz_detail()` in `capstone-rs/src/test.rs` (you can see similar tests for other archs).

**Review by tmfink — CHANGES_REQUESTED** [review]
thanks for the change!

File: capstone-rs/src/arch/sysz.rs (hunk 2 above)
Comment (GarntS) [reply capstone-rs/src/arch/sysz.rs:1]: Added this test

**Comment by GarntS** [comment]
Hey - made the changes you requested. Not 100% sure how Github's PRs work.

**Comment by tmfink** [comment]
Thanks for the contribution!

**Pull request merged**